lib/hls/hls_parser.js

```diff
@@ -2004,23 +2078,58 @@ shaka.hls.HlsParser = class {
     return null;
   }
 
+  /**
+   * Attempts to guess which codecs from the codecs list belong to a given
+   * content type. Does not assume a single codec is anything special, and does
+   * not throw if it fails to match.
+   *
+   * @param {string} uri
+   * @param {!Map.<string, string>} variables
+   * @return {string}
+   * @private
+   */
+  variableSubstitution_(uri, variables) {
+    let newUri = String(uri).replace(/%7B/g, '{').replace(/%7D/g, '}');
+
+    const uriVariables = newUri.match(/{\$\w*}/g);
+    if (uriVariables) {
+      for (const variable of uriVariables) {
+        // Note: All variables have the structure {$...}
+        const variableName = variable.slice(2, variable.length - 1);
+        const replaceValue = variables.get(variableName);
+        if (replaceValue) {
+          newUri = newUri.replace(variable, replaceValue);
+        } else {
+          shaka.log.error(uriVariables, variableName);
```

Comment:
Could you add some text to this log?  Without context, it's a bid odd.

Reply:
I have added more information to the log.